tiledb/api/src/query/strategy.rs

```diff
@@ -1598,22 +1624,7 @@ impl CellsStrategy {
     fn nrecords_limit(&self) -> Option<usize> {
         if let Some(schema) = self.schema.array_schema() {
             if !schema.allow_duplicates.unwrap_or(true) {
-                /*
```

Comment:
1. The iteration that is removed is essentially the same as the body of `schema.domain.num_cells()`
2. the issue that this comment mentions is resolved by code that has been written after this comment (in `CellStrategySchema::new_field_tree` near the comment `dimension coordinates must be unique, generate them first`)